libs/ledger-live-common/src/families/cardano/logic.ts

```diff
@@ -0,0 +1,442 @@
+import {
+  CARDANO_TESTNET_CURRENCY_ID,
+  STAKING_ADDRESS_INDEX,
+  TTL_GAP,
+} from "./constants";
+
+import {
+  utils as TyphonUtils,
+  types as TyphonTypes,
+  address as TyphonAddress,
+} from "@stricahq/typhonjs";
+import {
+  AddressType,
+  AssetGroup,
+  TxInput,
+  TxOutput,
+  TxOutputDestination,
+  TxOutputDestinationType,
+} from "@cardano-foundation/ledgerjs-hw-app-cardano";
+import { str_to_path } from "@cardano-foundation/ledgerjs-hw-app-cardano/dist/utils";
+import {
+  BipPath,
+  PaymentChain,
+  PaymentCredential,
+  StakeChain,
+  StakeCredential,
+  Token,
+} from "./types";
+import { Bip32PublicKey } from "@stricahq/bip32ed25519";
+import bs58 from "bs58";
+import BigNumber from "bignumber.js";
+import { getNetworkParameters } from "./networks";
+import { CryptoCurrency, OperationType } from "../../types";
+import groupBy from "lodash/groupBy";
+import { APITransaction } from "./api/api-types";
+
+/**
+ *  returns BipPath object with account, chain and index field for cardano
+ *
+ * @param {string} path
+ */
+export function getBipPathFromString(path: string): BipPath {
+  const regEx = new RegExp(/^1852'\/1815'\/(\d*)'\/([012])\/(\d*)/);
+  const result = path.match(regEx);
+  if (result == null) {
+    throw new Error("Invalid derivation path");
+  }
+  return getBipPath({
+    account: parseInt(result[1]),
+    chain: parseInt(result[2]),
+    index: parseInt(result[3]),
+  });
+}
+
+/**
+ * returns complete bipPath with purpose, coin, account, chain and index for cardano
+ */
+export function getBipPath({
+  account,
+  chain,
+  index,
+}: {
+  account: number;
+  chain: PaymentChain | StakeChain;
+  index: number;
+}): BipPath {
+  return {
+    purpose: 1852,
+    coin: 1815,
```

Comment:
You should set these in `constants.ts` and replace all uses in this file with the constants.